programs/drift/src/instructions/user.rs

```diff
@@ -1362,6 +1372,137 @@ pub fn handle_place_and_make_perp_order<'c: 'info, 'info>(
     Ok(())
 }
 
+pub fn place_swift_taker_order<'c: 'info, 'info>(
+    taker_key: Pubkey,
+    taker: &mut RefMut<User>,
+    taker_order_params_message: SwiftOrderParamsMessage,
+    ix_sysvar: &AccountInfo<'info>,
+    sig: [u8; 64],
+    perp_market_map: &PerpMarketMap,
+    spot_market_map: &SpotMarketMap,
+    oracle_map: &mut OracleMap,
+    state: &State,
+) -> Result<bool> {
+    // Authenticate the signature
+    let ix_idx = load_current_index_checked(ix_sysvar)?;
+    validate!(
+        ix_idx > 0,
+        ErrorCode::InvalidVerificationIxIndex,
+        "instruction index must be greater than 0"
+    )?;
+    let ix: Instruction = load_instruction_at_checked(ix_idx as usize - 1, ix_sysvar)?;
+    verify_ed25519_ix(
+        &ix,
+        &taker.authority.to_bytes(),
+        &taker_order_params_message.clone().try_to_vec()?,
+        &sig,
+    )?;
+
+    // Verify inputs and validate market index
+    taker_order_params_message.verify_all_same_market_indexes()?;
+    let market_index = taker_order_params_message.market_index;
+    let taker_order_params = taker_order_params_message.swift_order_params;
+
+    let clock = &Clock::get()?;
+    controller::repeg::update_amm(market_index, perp_market_map, oracle_map, state, clock)?;
+
+    // Use the first matching taker order param
+    let matching_taker_order_params_ref = taker_order_params.iter().find(|order_params| {
+        order_params.order_type != OrderType::TriggerLimit
+            && order_params.order_type != OrderType::TriggerMarket
+    });
+
+    if matching_taker_order_params_ref.is_none() {
+        msg!("No matching taker order params found");
+        // Place any matching orders and non-matching orders
+        return Ok(false);
+    }
+
+    let matching_taker_order_params = *matching_taker_order_params_ref.unwrap();
+
+    // Place taker order
+    let taker_order_sent = controller::orders::place_swift_perp_order(
+        state,
+        taker,
+        taker_key,
+        perp_market_map,
+        spot_market_map,
+        oracle_map,
+        clock,
+        matching_taker_order_params,
+        PlaceOrderOptions::default(),
+    )?;
+
+    if taker_order_sent {
+        let non_matching_orders = taker_order_params
+            .iter()
+            .filter(|order_params| {
+                order_params.order_type == OrderType::TriggerLimit
```

Comment:
i feel like these being outta order and having their own next order id is awkward.

I think it'd be cleaner if:
1) we only set expected_order_id once for bundle and its not optional
2) make market order the first one in order
3) only allow additional orders be triggers